### PR TITLE
Add workflow that requires core maintainer approval when editing requirements paths

### PR DESCRIPTION
**Pull request by daniellok-db**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/daniellok-db/mlflow/pull/11125?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/11125/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 11125
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

This PR adds a workflow that fails if a non-core maintainer tries to approve a PR that touches the `skinny-requirements.txt` or `setup.py` files. The reason for this is because dependency conflicts can be hard to manage with downstream tasks, so it's important for us to at least be aware of potential incompatibilities.

### How is this PR tested?

- [ ] Existing unit/integration tests
- [ ] New unit/integration tests
- [x] Manual tests

Example failure: https://github.com/mlflow/mlflow/actions/runs/7898767754/job/21556841255
Example success: hopefully it happens when someone approves this PR

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [x] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [x] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

### CODE REVIEW

**Review by harupy — COMMENTED** [review]
test

**Review by harupy — DISMISSED** [review]
test

**Review by harupy — APPROVED** [review]
LGTM!

**Comment by harupy** [comment]
can we mark this workflow as a required check once this PR is merged?

File: .github/workflows/warn-on-editing-requirements.yml
Comment (harupy) [inline .github/workflows/warn-on-editing-requirements.yml:5]: I thought the `pull_request` event has a type like approved but it doesn't seem to have it

File: .github/workflows/warn-on-editing-requirements.yml
Comment (harupy) [inline .github/workflows/warn-on-editing-requirements.yml:21]: can we move the code to the js file?

Comment (harupy) [reply .github/workflows/warn-on-editing-requirements.yml:21]: I see, we might use require-core-maintainer-approval.js for a reason in the future

Comment (daniellok-db) [reply .github/workflows/warn-on-editing-requirements.yml:21]: ah yes that was my intention

**Pull request merged**
